### DISCUSSION TITLE
Wrong version/modpack being grabbed

### DISCUSSION THREAD

**Discussion by Specimen1334** [itzg/docker-minecraft-server#2550] [Q&A]
Hi there, I am fairly new to docker.
I am running it on OMV6, and attempting to set up a modded minecraft server.
I have got my compose file set up, but it keeps telling me in the logs that i havent specified a version of minecraft, that i am using type=curseforge, and that im after the modpack ATM7.
however I have specified version=1.20.1, type=forge and modpack=after-midnight
any ideas on why it is ignoring my commands, and failing to boot?

<details>
<summary> Compose File </summary>
version: "1.1"
services:
  minecraft-server:
    image: itzg/minecraft-server
    container_name: minecraft
    network_mode: host
    environment:
      - PUID=998
      - PGID=100
      - EULA=TRUE
      - TYPE=FORGE
      - CF_API_KEY="mykey"
      - CF_MOD_PAGE=https://legacy.curseforge.com/minecraft/modpacks/after-midnight
      - VERSION=1.20.1
      - MEMORY=4G
      - ANNOUNCE_PLAYER_ACHIEVEMENTS=true
      - TZ=Australia/Hobart
      - ENABLE_ROLLING_LOGS=true
      - SERVER_NAME=Mordhaheim
      - LEVEL_NAME=Mordhaheim
      - ALLOW_FLIGHT=true
      - PVP=false
    ports:
      - 25565:25565
    volumes:
      - /docker/docker-config/curseforge:/mods
      - /docker/docker-config/curseforge:/modpacks
      - /docker/docker-config/curseforge:/data
    restart: unless-stopped
</details>

<details>
<summary> snippet of log (it just repeats) </summary>
[init] Running as uid=1000 gid=1000 with /data as 'drwxr-xr-x 2 1000 1000 4096 Dec  1 11:20 /data'
[init] WARNING for the old TYPE=CURSEFORGE mechanism it is best to set VERSION to a specific value
[init] Resolved version given LATEST into 1.20.2
[init] Looking for Feed-The-Beast / CurseForge server modpack.
[init] FTB server modpack /modpacks/ATM7-0.4.32-server.zip not found.
[init] Running as uid=1000 gid=1000 with /data as 'drwxr-xr-x 2 1000 1000 4096 Dec  1 11:20 /data'
[init] WARNING for the old TYPE=CURSEFORGE mechanism it is best to set VERSION to a specific value
[init] Resolved version given LATEST into 1.20.2
[init] Looking for Feed-The-Beast / CurseForge server modpack.
[init] FTB server modpack /modpacks/ATM7-0.4.32-server.zip not found.
[init] Running as uid=1000 gid=1000 with /data as 'drwxr-xr-x 2 1000 1000 4096 Dec  1 11:20 /data'
[init] WARNING for the old TYPE=CURSEFORGE mechanism it is best to set VERSION to a specific value
[init] Resolved version given LATEST into 1.20.2
[init] Looking for Feed-The-Beast / CurseForge server modpack.
[init] FTB server modpack /modpacks/ATM7-0.4.32-server.zip not found.
</details>

**Comment by itzg** [comment]
That can't be the actual compose file you're using since it uses `TYPE=FORGE`. Please post the correct file and I can give further advice from there. 

Also, I really don't want to provide further support for the old CurseForge mechanism. Please switch to https://docker-minecraft-server.readthedocs.io/en/latest/types-and-platforms/mod-platforms/auto-curseforge/

**Reply by Specimen1334** [reply]
I only have 1 compose file, and its creating two containers everytime, one sits there constantly copying and interpolating, its been over 24 hours of restarting, copying, interpolating, and the other has all that weird information saying im using type curseforge etc which doesnt make sense, because its not the settings I picked.
but yeah I only have a single compose file.

**Comment by itzg** [comment]
I am unable to provide further support until you sort out what is running and not. It sounds like misunderstood docker compose and started more before stopping others. 

I'd also recommend removing the restart policy 

```
restart: unless-stopped
```

until you have a system configured the way you want.

**Reply by Specimen1334** [reply]
I see what you mean, I have stopped the second entry, that was giving me the weird logs.
The other one in its boot cycle shows me the correct information.
It just keeps copying and interpolating endlessly though, and keeps copying the mods deeper and deeper. so itl go /mods, then to /mods/mods, then mods/mods/mods etc you get the idea. down about 8 layers deep then it restarts. I will remove the restart condition, it is still listed in the file given below.

<details>
<summary> Compose File </summary>
version: "1.2"
services:
  minecraft-server:
    image: itzg/minecraft-server
    container_name: minecraft
    network_mode: host
    environment:
      # Environment variables for configuration
      - PUID=998
      - PGID=100
      - EULA=TRUE
      - TYPE=AUTO_CURSEFORGE
      - CF_API_KEY="key"
      - CF_SLUG=after-midnight
      - VERSION=1.20.1
      - MEMORY=4G
      - ANNOUNCE_PLAYER_ACHIEVEMENTS=true
      - TZ=Australia/Hobart
      - ENABLE_ROLLING_LOGS=true
      - SERVER_NAME=Mordhaheim
      - LEVEL_NAME=Mordhaheim
      - ALLOW_FLIGHT=true
      - PVP=false
      - OVERRIDE_SERVER_PROPERTIES=true
    ports:
      # Mapping host port 25565 to container port 25565
      - 25565:25565
    volumes:
      # Mounting directories for mods, modpacks, and data
      - /docker/docker-config/curseforge:/mods
      - /docker/docker-config/curseforge:/modpacks
      - /docker/docker-config/curseforge:/data
    restart: unless-stopped
    
    </details

**Reply by itzg** [reply]
Can you please use code block formatting? It is very hard to read your compose file and only just now noticed your mounts are quite wrong with all three using the same source path:

```
- /docker/docker-config/curseforge:/mods 
- /docker/docker-config/curseforge:/modpacks 
- /docker/docker-config/curseforge:/data
```
